host-controller/src/main/java/org/jboss/as/host/controller/operations/HttpManagementAddHandler.java

```diff
@@ -127,10 +125,15 @@ static void populateHostControllerInfo(final LocalHostControllerInfoImpl hostCon
         hostControllerInfo.setAllowedOrigins(HttpManagementResourceDefinition.ALLOWED_ORIGINS.unwrap(context, model));
     }
 
-    public static void installHttpManagementServices(final RunningMode runningMode, final ServiceTarget serviceTarget, final LocalHostControllerInfo hostControllerInfo,
-                                                     final HostControllerEnvironment environment,
-                                                     boolean onDemand, boolean httpUpgrade, final ServiceRegistry serviceRegistry, final OptionMap options) {
+    static void installHttpManagementServices(final HostControllerEnvironment environment, final LocalHostControllerInfo hostControllerInfo, final OperationContext context,
+                                              final ModelNode model) throws OperationFailedException {
 
+        boolean httpUpgrade = HttpManagementResourceDefinition.HTTP_UPGRADE_ENABLED.resolveModelAttribute(context, model).asBoolean();
+        boolean onDemand = context.isBooting();
+        OptionMap options = createConnectorOptions(context, model);
+        ServiceRegistry serviceRegistry = context.getServiceRegistry(false);
```

Comment:
This is a pre-existing issue, but the param should be 'true', not 'false'. The caller intends to modify state.